http://sourceware.org/ml/binutils/2006-02/msg00127.html

Date: Wed, 8 Feb 2006 16:12:53 -0800
From: "H. J. Lu" <hjl@lucon.org>
To: Jeremy Brown <jhbrown@itasoftware.com>, binutils@sourceware.org
Subject: Re: ld and GNU/Linux ld.so disagree on dependent library search order (?)

On Thu, Feb 09, 2006 at 09:51:20AM +1030, Alan Modra wrote:
> On Tue, Feb 07, 2006 at 08:42:22AM -0800, H. J. Lu wrote:
> > 	PR ld/2290
> > 	* emultempl/elf32.em (gld${EMULATION_NAME}_after_open): Call
> > 	gld${EMULATION_NAME}_check_ld_so_conf before checking default
> > 	search directories for DT_NEEDED entries.
> 
> OK.  A NEWS entry would be nice too.
> 

I am going to check this in shortly.

Thanks.


H.J.
----
2006-02-07  H.J. Lu  <hongjiu.lu@intel.com>

	PR ld/2290
	* emultempl/elf32.em (gld${EMULATION_NAME}_after_open): Call
	gld${EMULATION_NAME}_check_ld_so_conf before checking default
	search directories for DT_NEEDED entries.

--- ld/emultempl/elf32.em.ld.so	2005-12-08 07:06:38.000000000 -0800
+++ ld/emultempl/elf32.em	2006-02-08 16:08:55.000000000 -0800
@@ -912,6 +912,17 @@ cat >>e${EMULATION_NAME}.c <<EOF
 
 EOF
 fi
+if [ "x${USE_LIBPATH}" = xyes ] ; then
+  case ${target} in
+    *-*-linux-gnu*)
+      cat >>e${EMULATION_NAME}.c <<EOF
+	  if (gld${EMULATION_NAME}_check_ld_so_conf (l->name, force))
+	    break;
+EOF
+    # Linux
+    ;;
+  esac
+fi
 cat >>e${EMULATION_NAME}.c <<EOF
 	  len = strlen (l->name);
 	  for (search = search_head; search != NULL; search = search->next)
@@ -930,17 +942,6 @@ cat >>e${EMULATION_NAME}.c <<EOF
 	  if (search != NULL)
 	    break;
 EOF
-if [ "x${USE_LIBPATH}" = xyes ] ; then
-  case ${target} in
-    *-*-linux-gnu*)
-      cat >>e${EMULATION_NAME}.c <<EOF
-	  if (gld${EMULATION_NAME}_check_ld_so_conf (l->name, force))
-	    break;
-EOF
-    # Linux
-    ;;
-  esac
-fi
 cat >>e${EMULATION_NAME}.c <<EOF
 	}
 

